es6/timeline.js

```diff
@@ -10,31 +10,64 @@ const page_start_time = Date.now()
 const offset = (time) =>
     time - page_start_time
 
+function getCssProperty(elmId, property){
+   var elem = document.getElementById(elmId);
+   return elem ?
+          window.getComputedStyle(elem,null).getPropertyValue(property)
+          : 0
+}
+
+const current_width = () => {
+    const elem = document.getElementById("animations_container")
+    return elem ? elem.getBoundingClientRect().width : 1
+}
+
+const current_frame_position = () => {
+    const left_property = getCssProperty("current_frame", "left")
+    const left = typeof(left_property) === "number" ?
+                 left_property
+                 : Number(left_property.replace("px", ""))
+    return left
+}
 
-const AnimRow = (anim, idx, scale) => {
+const AnimRow = (anim, idx, scale, warped_time) => {
```

Comment:
This returns a react component, so it needs to take an object of kwargs, not args:

```javascript
const AnimRow = ({anim, idx, scale, warped_time}) => {
```